### PR TITLE
ECL-789 - Return success page error

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala [hunk 1]
```diff
@@ -45,11 +45,11 @@ class ReturnSubmittedController @Inject() (
     with I18nSupport
     with ErrorHandler {
 
-  def onPageLoad: Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+  def onPageLoad: Action[AnyContent] = authorise.async { implicit request =>
     val chargeReference: Option[String] = request.session.get(SessionKeys.ChargeReference)
     val amountDue: BigDecimal           = BigDecimal(request.session(SessionKeys.AmountDue))
-    val obligationDetails               = request.eclReturn.obligationDetails
-    val email                           = request.eclReturn.contactEmailAddress
+    val obligationDetails               = ObligationDetails.read(request.session.get(SessionKeys.ObligationDetails))
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala:51]: Do we want to use the valueOrError on here, like registration has?

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/StartController.scala [hunk 2]
```diff
@@ -127,6 +135,13 @@ class StartController @Inject() (
       )
     )
 
+  private def addPeriodKeyToSession(periodKey: String)(implicit
```
Comment (TheDanielDoyle) [inline app/uk/gov/hmrc/economiccrimelevyreturns/controllers/StartController.scala:138]: Can this be a more general method on the Base controller?

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala (hunk 1 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala:49]: Resolved, all of these fields are handled in for comprehension below.

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/StartController.scala (hunk 2 above)
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyreturns/controllers/StartController.scala]: Done

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**